File: dbflow-processor/src/main/java/com/raizlabs/android/dbflow/processor/definition/column/WrapperColumnAccess.java
package com.raizlabs.android.dbflow.processor.definition.column;

/**
 * Description:
 */
public abstract class WrapperColumnAccess extends BaseColumnAccess {

    protected ColumnDefinition columnDefinition;
    protected BaseColumnAccess existingColumnAccess;

    public WrapperColumnAccess(ColumnDefinition columnDefinition) {
        this.existingColumnAccess = columnDefinition.getColumnAccess();
        this.columnDefinition = columnDefinition;
    }

    public BaseColumnAccess getExistingColumnAccess() {
        return existingColumnAccess;
    }

}

File: dbflow-processor/src/main/java/com/raizlabs/android/dbflow/processor/definition/column/WrapperColumnAccess.kt
package com.raizlabs.android.dbflow.processor.definition.column

/**
 * Description:
 */
abstract class WrapperColumnAccess(protected var columnDefinition: ColumnDefinition)
: BaseColumnAccess() {
    var existingColumnAccess: BaseColumnAccess
        protected set

    init {
        this.existingColumnAccess = columnDefinition.columnAccess
    }

}
